### PR TITLE
docs: enable docstring linting and fix errors

### LINKED ISSUE #253: Lint documentation
- [x] Install and configure flake8-docstring
- [x] Fill out missing documentation in public APIs and disable where appropriate
- [x] Fix errors in existing documentation and style

### PR DESCRIPTION
**Pull request by malandis**
Enable docstring linting and fix errors.

Closes #253

### CODE REVIEW

File: src/momento/config/transport/transport_strategy.py [hunk 1]
```diff
@@ -13,24 +13,34 @@ class TransportStrategy(ABC):
...
 
     @abstractmethod
     def with_grpc_configuration(self, grpc_configuration: GrpcConfiguration) -> TransportStrategy:
-        """Copy constructor for overriding the gRPC configuration
+        """Copy constructor for overriding the gRPC configuration.
 
-        :param grpc_configuration: GrpcConfiguration
-        :return: a new TransportStrategy with the specified gRPC config.
+        Args:
+            grpc_configuration (GrpcConfiguration): gRPC configuration.
```
Comment (schwern) [inline src/momento/config/transport/transport_strategy.py]:
```suggestion
            grpc_configuration (GrpcConfiguration): the new gRPC configuration to use
```

File: src/momento/config/transport/transport_strategy.py [hunk 2]
```diff
@@ -13,24 +13,34 @@ class TransportStrategy(ABC):
...
+
+        Returns:
+            TransportStrategy: a new TransportStrategy with the specified gRPC config.
         """
         pass
 
     @abstractmethod
     def with_client_timeout(self, client_timeout: timedelta) -> TransportStrategy:
-        """Copy constructor to update the client-side timeout
+        """Copy constructor to update the client-side timeout.
```
Comment (schwern) [inline src/momento/config/transport/transport_strategy.py:37]:
```suggestion
        """Copies the TransportStrategy and updates the copy's client-side timeout.
```

File: src/momento/config/transport/transport_strategy.py [hunk 3]
```diff
@@ -13,24 +13,34 @@ class TransportStrategy(ABC):
...
+
+        Args:
+            client_timeout (timedelta): timedelta representing the new client timeout value.
```
Comment (schwern) [inline src/momento/config/transport/transport_strategy.py:40]:
```suggestion
            client_timeout (timedelta): the new client-side timeout.
```

File: src/momento/config/transport/transport_strategy.py [hunk 4]
```diff
@@ -13,24 +13,34 @@ class TransportStrategy(ABC):
...
 
-        :param client_timeout: timedelta representing the new client timeout value
-        :return: a new TransportStrategy with the specified client timeout.
+        Returns:
+            TransportStrategy: a new TransportStrategy with the specified client timeout.
```
Comment (schwern) [inline src/momento/config/transport/transport_strategy.py:43]:
```suggestion
            TransportStrategy: the new TransportStrategy.
```

File: src/momento/config/configuration.py [hunk 5]
```diff
@@ -24,28 +24,44 @@ def with_client_timeout(self, client_timeout: timedelta) -> Configuration:
...
-        """Copy constructor for overriding TransportStrategy
+        """Copy constructor for overriding TransportStrategy.
+
+        Args:
+            transport_strategy (TransportStrategy): the new TransportStrategy.
 
-        :param transport_strategy: TransportStrategy
-        :return: a new Configuration object with the specified TransportStrategy
+        Returns:
+            Configuration: a new Configuration object with the specified TransportStrategy.
```
Comment (schwern) [inline src/momento/config/configuration.py:53]:
```suggestion
            Configuration: the new Configuration with the specified TransportStrategy.
```

**Review by schwern — COMMENTED** [review]
Still reviewing, but here's some suggestions.

File: src/momento/internal/aio/_add_header_client_interceptor.py [hunk 6]
```diff
@@ -51,11 +51,12 @@ async def intercept_unary_unary(
...
-    when we inject our own metadata onto request object. This was mainly done as temporary fix after we observed
-    ddtrace grpc client interceptor passing client_call_details.metadata as a list instead of a grpc.aio.Metadata
-    object. See this ticket for follow-up actions to come back in and address this longer term:
+    """Defensive function meant to handle inbound gRPC client request objects.
+
+    Makes sure we can handle properly when we inject our own metadata onto request object.
+    This was mainly done as temporary fix after we observed ddtrace grpc client interceptor passing
+    client_call_details.metadata as a list instead of a grpc.aio.Metadata object.
+    See this ticket for follow-up actions to come back in and address this longer term:
         https://github.com/momentohq/client-sdk-python/issues/149
```
Comment (schwern) [inline src/momento/internal/aio/_add_header_client_interceptor.py:60]: Should this be in the docs? Or should it be a comment?

File: src/momento/internal/aio/_add_header_client_interceptor.py [hunk 7]
```diff
@@ -51,11 +51,12 @@ async def intercept_unary_unary(
...
-    """
-    Defensive function meant to handle inbound grpc client request objects and make sure we can handle properly
-    when we inject our own metadata onto request object. This was mainly done as temporary fix after we observed
-    ddtrace grpc client interceptor passing client_call_details.metadata as a list instead of a grpc.aio.Metadata
-    object. See this ticket for follow-up actions to come back in and address this longer term:
+    """Defensive function meant to handle inbound gRPC client request objects.
+
+    Makes sure we can handle properly when we inject our own metadata onto request object.
+    This was mainly done as temporary fix after we observed ddtrace grpc client interceptor passing
+    client_call_details.metadata as a list instead of a grpc.aio.Metadata object.
```
Comment (schwern) [inline src/momento/internal/aio/_add_header_client_interceptor.py:58]:
```suggestion
    Makes sure we can properly handle injecting our metadata into the gRPC request.
```

The rest seems like a comment.

File: src/momento/config/configurations.py [hunk 8]
```diff
@@ -20,23 +21,30 @@ def latest() -> Laptop:
 
 
 class InRegion:
-    """InRegion provides defaults suitable for an environment where your client is running in the same region as the
+    """Default for application running in the same region as the Momento service.
+
+    InRegion provides defaults suitable for an environment where your client is running in the same region as the
     Momento service.  It has more aggressive timeouts and retry behavior than the Laptop config.
     """
```
[the remark this answers is not in the document]
Comment (malandis) [reply src/momento/config/configurations.py:28]: This needs to be in the format:

one line summary.

description

File: src/momento/requests/collection_ttl.py [hunk 9]
```diff
@@ -82,19 +84,20 @@ def refresh_ttl_if_provided(ttl: Optional[timedelta] = None) -> CollectionTtl:
         return CollectionTtl(ttl=ttl, refresh_ttl=ttl is not None)
 
     def with_refresh_ttl_on_updates(self) -> CollectionTtl:
-        """Specifies that the TTL for the collection should be refreshed when
-        the collection is modified.  (This is the default behavior.)
+        """Specifies the TTL for the collection be refreshed when the collection is modified.
+
+        NB: This is the default behavior.
```
Comment (schwern) [inline src/momento/requests/collection_ttl.py:89]: I had to look up what NB meant. ESL users will probably have trouble, too.

```suggestion
        This is the default behavior for a CollectionTTL.
```

File: src/momento/requests/collection_ttl.py [hunk 10]
```diff
@@ -82,19 +84,20 @@ def refresh_ttl_if_provided(ttl: Optional[timedelta] = None) -> CollectionTtl:
...
 
     def with_no_refresh_ttl_on_updates(self) -> CollectionTtl:
-        """Specifies that the TTL for the collection should not be refreshed
-        when the collection is modified.  Use this if you want to ensure
-        that your collection expires at the originally specified time, even
-        if you make modifications to the value of the collection.
+        """Specifies the TTL for the collection should not be refreshed when the collection is modified.
+
+        Use this if ou want to ensure that your collection expires at the originally specified time,
+        even if you make modifications to the value of the collection.
```
Comment (schwern) [inline src/momento/requests/collection_ttl.py:100]: I feel like limiting this to only when the values are modified will bite the user when we inevitably decide to apply refresh on delete or something. YMMV

```suggestion
        Use this to ensure your collection expires at the originally specified time,
        even if you make modifications to the collection.
```

File: src/momento/responses/data/set/fetch.py [hunk 11]
```diff
@@ -8,8 +8,9 @@
 
 
 class CacheSetFetchResponse(CacheResponse):
-    """Response type for a `set_fetch` request. Its subtypes are:
+    """Response type for a `set_fetch` request.
```
Comment (schwern) [inline src/momento/responses/data/set/fetch.py:11]:
```suggestion
    """Parent response type for a `set_fetch` request.
```

File: tests/conftest.py [hunk 12]
```diff
@@ -195,7 +195,7 @@ def bad_auth_token() -> str:
 
 @pytest.fixture(scope="session")
 def event_loop() -> Iterator[asyncio.AbstractEventLoop]:
-    """cf https://github.com/pytest-dev/pytest-asyncio#event_loop"""
+    """Cf https://github.com/pytest-dev/pytest-asyncio#event_loop."""
```
Comment (schwern) [inline tests/conftest.py:198]:
```suggestion
    """See also: https://github.com/pytest-dev/pytest-asyncio#event_loop."""
```

**Review by schwern — DISMISSED** [review]

File: src/momento/logs.py [hunk 13]
```diff
@@ -25,9 +23,9 @@ def add_logging_level(level_name: str, level_num: int, method_name: Optional[str
 
     To avoid accidental clobberings of existing attributes, this method will
     raise an `AttributeError` if the level name is already an attribute of the
-    `logging` module or if the method name is already present
+    `logging` module or if the method name is already present.
 
-    Example
+    Example::
```
Comment (nand4011) [inline src/momento/logs.py:28]: Make sure this renders properly without a blank line after. I had some trouble with example code not respecting newlines.

Comment (malandis) [reply src/momento/logs.py:28]: Looks good in VS Code.

Comment (malandis) [reply src/momento/logs.py:28]: Looks good in VS Code. I think the (pre-existing) dashes took care of that

File: src/momento/internal/aio/_scs_grpc_manager.py [hunk 14]
```diff
@@ -12,7 +12,7 @@
 
 
 class _ControlGrpcManager:
-    """Momento Internal."""
+    """Internal gRPC control maanger."""
```
Comment (pgautier404) [inline src/momento/internal/aio/_scs_grpc_manager.py:15]:
```suggestion
    """Internal gRPC control mananger."""
```

File: src/momento/internal/aio/_scs_grpc_manager.py [hunk 15]
```diff
@@ -31,7 +31,7 @@ def async_stub(self) -> control_client.ScsControlStub:
 
 
 class _DataGrpcManager:
-    """Momento Internal."""
+    """Internal gRPC data maanger."""
```
Comment (pgautier404) [inline src/momento/internal/aio/_scs_grpc_manager.py:34]:
```suggestion
    """Internal gRPC data mananger."""
```

File: src/momento/internal/synchronous/_scs_grpc_manager.py [hunk 16]
```diff
@@ -16,7 +16,7 @@
 
 
 class _ControlGrpcManager:
-    """Momento Internal."""
+    """Internal gRPC control maanger."""
```
Comment (pgautier404) [inline src/momento/internal/synchronous/_scs_grpc_manager.py:19]:
```suggestion
    """Internal gRPC control mananger."""
```

File: src/momento/internal/synchronous/_scs_grpc_manager.py [hunk 17]
```diff
@@ -35,7 +35,7 @@ def stub(self) -> control_client.ScsControlStub:
 
 
 class _DataGrpcManager:
-    """Momento Internal."""
+    """Internal gRPC data maanger."""
```
Comment (pgautier404) [inline src/momento/internal/synchronous/_scs_grpc_manager.py:38]:
```suggestion
    """Internal gRPC data mananger."""
```

**Review by schwern — APPROVED** [review]

**Review by schwern — APPROVED** [review]

File: src/momento/responses/data/list/concatenate_back.py [hunk 18]
```diff
@@ -6,8 +6,9 @@
 
 
 class CacheListConcatenateBackResponse(CacheResponse):
-    """Response type for a `list_concatenate_back` request. Its subtypes are:
+    """Response type for a `list_concatenate_back` request.
```
Comment (pgautier404) [inline src/momento/responses/data/list/concatenate_back.py:9]: Nit: maybe change to "Parent response type . . ." for consistency? It looks like a large majority of these docs follow that convention except for a few list responses.

File: src/momento/responses/data/list/concatenate_back.py [hunk 19]
```diff
@@ -26,8 +27,9 @@ class Success(CacheListConcatenateBackResponse):
         """The number of values in the list after this concatenation"""
 
     class Error(CacheListConcatenateBackResponse, ErrorResponseMixin):
-        """Indicates an error occured in the request:
+        """Indicates an error occured in the request.
```
Comment (pgautier404) [inline src/momento/responses/data/list/concatenate_back.py:30]:
```suggestion
        """Indicates an error occurred in the request.
```

File: src/momento/responses/data/list/concatenate_front.py [hunk 20]
```diff
@@ -26,8 +27,9 @@ class Success(CacheListConcatenateFrontResponse):
         """The number of values in the list after this concatenation"""
 
     class Error(CacheListConcatenateFrontResponse, ErrorResponseMixin):
-        """Indicates an error occured in the request:
+        """Indicates an error occured in the request.
```
Comment (pgautier404) [inline src/momento/responses/data/list/concatenate_front.py:30]:
```suggestion
        """Indicates an error occurred in the request.
```

Comment (pgautier404) [reply src/momento/responses/data/list/concatenate_front.py:30]: Looks like this typo is pretty pervasive and could use a batch find and replace.

File: src/momento/simple_cache_client.py [hunk 21]
```diff
@@ -583,12 +575,10 @@ def list_pop_back(self, cache_name: str, list_name: str) -> CacheListPopBackResp
         Returns:
             CacheListPopBackResponse:
         """
-
         return self._data_client.list_pop_back(cache_name, list_name)
 
     def list_pop_front(self, cache_name: str, list_name: str) -> CacheListPopFrontResponse:
-        """
-        Gets removes and returns the first value from the list.
+        """Gets removes and returns the first value from the list.
```
Comment (pgautier404) [inline src/momento/simple_cache_client.py:581]:
```suggestion
        """Gets, removes, and returns the first value from the list.
```

File: src/momento/simple_cache_client_async.py [hunk 22]
```diff
@@ -571,12 +565,10 @@ async def list_length(self, cache_name: str, list_name: str) -> CacheListLengthR
         Returns:
             CacheListLengthResponse:
         """
-
         return await self._data_client.list_length(cache_name, list_name)
 
     async def list_pop_back(self, cache_name: str, list_name: str) -> CacheListPopBackResponse:
-        """
-        Gets removes and returns the last value from the list.
+        """Gets removes and returns the last value from the list.
```
Comment (pgautier404) [inline src/momento/simple_cache_client_async.py:571]:
```suggestion
        """Gets, removes, and returns the last value from the list.
```

File: src/momento/simple_cache_client_async.py [hunk 23]
```diff
@@ -585,12 +577,10 @@ async def list_pop_back(self, cache_name: str, list_name: str) -> CacheListPopBa
         Returns:
             CacheListPopBackResponse:
         """
-
         return await self._data_client.list_pop_back(cache_name, list_name)
 
     async def list_pop_front(self, cache_name: str, list_name: str) -> CacheListPopFrontResponse:
-        """
-        Gets removes and returns the first value from the list.
+        """Gets removes and returns the first value from the list.
```
Comment (pgautier404) [inline src/momento/simple_cache_client_async.py:583]:
```suggestion
        """Gets, removes, and returns the first value from the list.
```

File: src/momento/responses/data/list/concatenate_back.py (hunk 18 above)
Comment (malandis) [reply src/momento/responses/data/list/concatenate_back.py:9]: I will address this in a follow up PR

**Review by pgautier404 — APPROVED** [review]
LGTM!

**Pull request merged**